adobe/react-spectrum
# fix(#1118): ColorArea component

Closes  [React Spectrum GitHub Issue](https://github.com/adobe/react-spectrum/issue/1118)

## ✅ Pull Request Checklist:

- [x] Included link to corresponding [React Spectrum GitHub Issue](https://github.com/adobe/react-spectrum/issue/1118).

## 📝 Test Instructions:

1. Open https://reactspectrum.blob.core.windows.net/reactspectrum/d02b605cb71505dbf074eb9f3b54c9174538bc8c/storybook/index.html?path=/story/colorarea--default-hsb-xchannel-saturation-ychannel-brightness
2. Verify that with mouse press on the ColorArea, the ColorThumb moves and the color updates appropriately.
3. The the input element for the horizontal channel should receive keyboard focus and the ColorThumb should display its focused state.
4. Use the LEFT/RIGHT keys to increment or decrement the value of the horizontal channel or UP/DOWN keys to increment or decrement the value of the vertical channel.
5. Use PageUp/PageDown or Shift+UP/Shift+DOWN keys to keys to increment or decrement the value of the vertical channel or HOME/END or Shift+LEFT/Shift+RIGHT to increment or decrement the value of the horizontal channel by snapping to the nearest default step appropriate to the channel. 
  a. Hue increments/decrements by 15, 
  b. RGB increments/decrements by 16, 
  c. and Saturation, Lightness or Brightness increment/decrement by 10.
6. Verify that the gradient changes when the external "z" channel value changes.
7. Repeat for each of the ColorArea channel variations.
8. Verify that RTL languages reverse the orientation of the horizontal axis.
9. Using a screen reader like VoiceOver, verify that incrementing or decrementing in the vertical direction announces the value of the vertical channel first, and that incrementing or decrementing in the horizontal direction announces the value of the horizontal channel first. After a delay, the title attribute should be announced as the full value for the color space.

## 🧢 Your Project:

Adobe/Accessibility
